Remotion/Web/ClientScript/Scripts/ListMenu.ts

```diff
@@ -0,0 +1,300 @@
+        // Do not render empty diagnostic metadata attributes
+        $.each(itemInfo.DiagnosticMetadataForCommand,
+          function (key, value) {
+            if (value === "" || value === null)
+            {
+              delete itemInfo.DiagnosticMetadataForCommand[key];
+            }
+          });
+
+        itemInfo.DiagnosticMetadataForCommand['data-is-disabled'] = isEnabled ? 'false' : 'true';
+
+        $(anchor).attr(itemInfo.DiagnosticMetadataForCommand);
+      }
+    }
+  }
+
+  public static OnKeyDown (event: JQueryKeyEventObject, $listMenu: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery ('$tablist', $listMenu);
+
+    var $menuItems = $listMenu.find ('a');
+
+    var oldMenuItemIndex = -1;
+
+    var selectedMenuItem = document.activeElement;
+    if (selectedMenuItem != null && TypeUtility.IsDefined (selectedMenuItem.tagName) && selectedMenuItem.tagName.toUpperCase() === 'A')
+    {
+      oldMenuItemIndex = $menuItems.index(selectedMenuItem);
+    }
+    else
+    {
+      for (var i = 0; i < $menuItems.length; i++)
+      {
+        // TODO RM-7686 Fix misspellings of tabIndex
+        if (($menuItems[i] as any).tabindex === 0)
+        {
+          oldMenuItemIndex = i;
+          break;
+        }
+      }
+    }
+
+    var $oldMenuItem = null;
+    if (oldMenuItemIndex >= 0)
+      $oldMenuItem = $($menuItems[oldMenuItemIndex]);
+    var currentMenuItemIndex = Math.max (0, oldMenuItemIndex);
+
+    switch (event.keyCode)
+    {
+      case 9: // tab
+        // exit tab strip
+        return;
+      case 13: //enter
+      case 32: //space
+        {
+          event.preventDefault();
+          event.stopPropagation();
+
+          if (currentMenuItemIndex >= 0)
+          {
+            let $newMenuItem = $($menuItems[currentMenuItemIndex]);
+            ListMenu_UpdateFocus ($newMenuItem, $oldMenuItem);
+
+            $newMenuItem[0].click();
+          }
+
+          return;
+        }
+      case 37: // left arrow
+      case 38: // up arrow
+        {
+          event.preventDefault();
+          event.stopPropagation();
+
+          if (currentMenuItemIndex > 0)
+            currentMenuItemIndex--;
+          else
+            currentMenuItemIndex = $menuItems.length - 1;
+
+          let $newMenuItem = $ ($menuItems[currentMenuItemIndex]);
+          ListMenu_UpdateFocus ($newMenuItem, $oldMenuItem);
+
+          return;
+        }
+      case 39: // right arrow
+      case 40: // down arrow
+        {
+          event.preventDefault();
+          event.stopPropagation();
+
+          if (currentMenuItemIndex < $menuItems.length - 1)
+            currentMenuItemIndex++;
+          else
+            currentMenuItemIndex = 0;
+
+          var $newMenuItem = $($menuItems[currentMenuItemIndex]);
+          ListMenu_UpdateFocus ($newMenuItem, $oldMenuItem);
+
+          return;
+        }
+    }
+  }
+
+  public static UpdateFocus ($newMenuItem: JQuery, $oldMenuItem: Nullable<JQuery>): void
```

Comment:
Is this only needed from within the own class or also from other places?

Reply:
I only found references to this function within this file, no idea about downstream projects though, that's why I left the global function as-is.

Reply:
we may assume they're unused.